app/assets/javascripts/discourse/tests/integration/components/float-kit/d-menu-test.js

```diff
@@ -0,0 +1,186 @@
+  });
+
+  test("<:trigger>", async function (assert) {
+    await render(
+      hbs`<DMenu @inline={{true}}><:trigger>label</:trigger></DMenu />`
+    );
+
+    assert.dom(".d-menu__trigger").containsText("label");
+  });
+
+  test("<:content>", async function (assert) {
+    await render(
+      hbs`<DMenu @inline={{true}}><:content>content</:content></DMenu />`
+    );
+
+    await open();
+
+    assert.dom(".d-menu").containsText("content");
+  });
+
+  test("content role attribute", async function (assert) {
+    await render(hbs`<DMenu @inline={{true}} @label="label"  />`);
+
+    await open();
+
+    assert.dom(".d-menu").hasAttribute("role", "dialog");
+  });
+
+  test("@component", async function (assert) {
+    this.component = DDefaultToast;
+
+    await render(
+      hbs`<DMenu @inline={{true}} @label="test" @component={{this.component}} @model={{hash message="content"}}/>`
+    );
+
+    await open();
+
+    assert.dom(".d-menu").containsText("content");
+
+    await click(".d-menu .btn");
+
+    assert.dom(".d-menu").doesNotExist();
+  });
+
+  test("content aria-labelledby attribute", async function (assert) {
+    await render(hbs`<DMenu @inline={{true}} @label="label"  />`);
+
+    await open();
+
+    assert.strictEqual(
+      document.querySelector(".d-menu__trigger").id,
+      document.querySelector(".d-menu").getAttribute("aria-labelledby")
+    );
+  });
+
+  test("@closeOnEscape", async function (assert) {
+    await render(
+      hbs`<DMenu @inline={{true}} @label="label" @closeOnEscape={{true}}  />`
+    );
+    await open();
+    await triggerKeyEvent(document.activeElement, "keydown", "Escape");
+
+    assert.dom(".d-menu").doesNotExist();
+
+    await render(
+      hbs`<DMenu @inline={{true}} @label="label" @closeOnEscape={{false}}  />`
+    );
+    await open();
+    await triggerKeyEvent(document.activeElement, "keydown", "Escape");
+
+    assert.dom(".d-menu").exists();
+  });
+
+  test("@closeOnClickOutside", async function (assert) {
+    await render(
+      hbs`<span class="test">test</span><DMenu @inline={{true}} @label="label" @closeOnClickOutside={{true}}  />`
+    );
+    await open();
+    await click(".test");
+
+    assert.dom(".d-menu").doesNotExist();
+
+    await render(
+      hbs`<span class="test">test</span><DMenu @inline={{true}} @label="label" @closeOnClickOutside={{false}}  />`
+    );
+    await open();
+    await click(".test");
+
+    assert.dom(".d-menu").exists();
+  });
+
+  test("@maxWidth", async function (assert) {
+    await render(
+      hbs`<DMenu @inline={{true}} @label="label" @maxWidth={{20}}  />`
+    );
+    await open();
+
+    assert.ok(
+      find(".d-menu").getAttribute("style").includes("max-width: 20px;")
+    );
+  });
+
+  test("applies position", async function (assert) {
+    await render(hbs`<DMenu @inline={{true}} @label="label"  />`);
+    await open();
+
+    assert.ok(find(".d-menu").getAttribute("style").includes("left: "));
```

Comment:
All the attribute assertions could also use `qunit-dom`-ification 😃 like:

```suggestion
    assert.dom(".d-menu").hasAttribute("style", /left: /);
```